app/views/users/nbp_wallet/connect.html.slim

```diff
@@ -0,0 +1,21 @@
+.header.mb-5
+  .std-heading
+    i.fa-solid.fa-wallet
+    =< t('.header')
+
+.row
+  .col-8.col-md-4.col-lg-3
+    img.nbp_wallet_qr_code[src=@template.qr_code_path data-template-validity=@template.validity_period alt=t('.qr_code_alt_text')]
+    = button_tag class: 'btn btn-primary regenerate-qr-code-button w-100 mt-3' do
+      = t('.regenerate_code')
+    = link_to nbp_wallet_cancel_users_path, class: 'btn btn-primary w-100 mt-3'
```

Comment:
I was wondering whether we should use the actual "logout" mechanism we already have:

```suggestion
    = link_to destroy_user_session_path, method: :delete, class: 'btn btn-primary w-100 mt-3'
```

This approach has at least one drawback, and that is the removal of the custom flash message shown when canceling the registration. Apart from that, it ensures we have only one main code path, avoid duplications, and also initiate a SAML logout if supported by the provider (which is not the case for the mock provider, but for the actual NBP provider).

Reply:
I tested locally and ```destroy_user_session_path``` does not seem to remove saml_uid and omniauth_provider from the session. I can still navigate to ```http://localhost:3000/users/nbp_wallet/connect``` after "signing out".

Reply:
Yes, that could happen with the Mock Provider.. Therefore, I worked on #1453 already.

The background is an error in our handling when Single Log-Out (SLO) is not provided. With CodeHarbor, SLO works as follows:
1. The user session is terminated first, i.e., logging the user out of their account. This is handled by devise and usually clears the session completely.
2. Then, the SAML SLO workflow is initialized, which redirects the user to the identity provider used during login. As a user can have multiple identity providers, we need to know which one (and some further SAML data). This is the data stored as `saml_uid`, `saml_session_index` and `omniauth_provider`.
3. The Identity Provider handles the logout request and _should_ redirect the user either to the next service or back to CodeHarbor.
4. Back on CodeHarbor, the SAML session is cleared, too, effectively finishing the logout request.

As you see, clearing the session first is not the smartest way to do regarding the SLO workflow, since some data is required in the second step. Hence, it would be enough to do clear the session in the forth steps. However, if the provider does not redirect back (which some won't) or something goes wrong, a user wouldn't have cleared their CodeHarbor session.

Therefore, we decided to "keep" the SAML session to the CodeHarbor session during the first step, allowing the remaining workflow to work. This data was removed during the forth step. At the time of implementation, every provider (including the SAML test IdP we used for development at that time) supported SLO, so that this worked. Now, the SAML Mock provider currently used doesn't support SLO, which effectively cut the process short and just "terminated" after the first step. My PR #1453 resolves this issue and ensures that a logout will clear the SAML session even if SLO is not supported.

Reply:
Okay, I removed the cancel route and switched to ```destroy_user_session_path```. We should test again when #1453 is merged.

Reply:
Sure. Feel free to review #1453. I'll keep this thread open so that we can then test that behavior again (and potentially write a test 🫣).

Reply:
@Mathis-Z Feel free to double-check this behavior too, so that we can resolve it :)

Reply:
I tested it locally again and it works as expected :+1: